### PR TITLE
refactor: Add StackParams type

### PR DESCRIPTION
**Pull request by gorandalum**
This type is meant to be used on every stack params specification. It both:
- Checks that each key in the stack param list ends with "Screen" or "Stack"
- Adds the CustomScreenParams to the mapped value type., which makes it
  possible to set transition when navigating.

### CODE REVIEW

File: src/stacks-hierarchy/RootStack.tsx [hunk 1]
```diff
@@ -392,19 +392,19 @@ export const RootStack = () => {
                 component={Root_AddPaymentMethodScreen}
               />
               <Stack.Screen
-                name="Root_ParkingViolationsSelect"
+                name="Root_ParkingViolationsSelectScreen"
                 component={Root_ParkingViolationsSelect}
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/RootStack.tsx:396]: Update the component names as well?

**Review by rosvik — APPROVED** [review]
Clean ✨

File: src/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_ZonePickerScreen.tsx [hunk 2]
```diff
@@ -43,8 +43,8 @@ export const TicketAssistant_ZonePickerScreen = ({
 
   const isApplicableOnSingleZoneOnly = false;
 
-  const fromTariffZone = route.params?.fromTariffZone ?? defaultTariffZone;
-  const toTariffZone = route.params?.toTariffZone ?? defaultTariffZone;
+  const fromTariffZone = defaultTariffZone;
+  const toTariffZone = defaultTariffZone;
```
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_ZonePickerScreen.tsx:47]: So this does not depend on any more of incoming parameters in the zone?

Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_ZonePickerScreen.tsx:47]: OK, got it! Seems like not related but good to have then I think.

**Review by jorelosorio — APPROVED** [review]
Good approach

**Pull request merged**